### PR TITLE
Add clean CAVLC B-frame decoder support.

### PR DESCRIPTION
**Pull request by xiaotianshicomcast**

### CODE REVIEW

File: codec/decoder/core/inc/decode_slice.h [hunk 1]
```diff
@@ -1,99 +1,103 @@
...
-void WelsBlockZero16x16_c (int16_t* block, int32_t stride);
-void WelsBlockZero8x8_c (int16_t* block, int32_t stride);
-
-} // namespace WelsDec
-
-#endif //WELS_DECODE_SLICE_H__
-
-
+/*!
+ * \copy
```
Comment (thaytan) [inline codec/decoder/core/inc/decode_slice.h]: This header is almost all whitespace change because of switching to Windows line endings. I'm not sure why that's happened just to this file and parse_mb_syn_cavlc.h and not others.

File: codec/decoder/core/src/mv_pred.cpp [hunk 2]
```diff
@@ -320,6 +320,10 @@ int32_t GetColocatedMb (PWelsDecoderContext pCtx, MbType& mbType, SubMbType& sub
   }
 
   MbType coloc_mbType = colocPic->pMbType[iMbXy];
+  if (coloc_mbType == MB_TYPE_SKIP) {
+    //This indicates the colocated MB is P SKIP MB
+    coloc_mbType |= MB_TYPE_16x16 | MB_TYPE_P0L0 | MB_TYPE_P1L0;
+  }
```
Comment (thaytan) [inline codec/decoder/core/src/mv_pred.cpp]: This change doesn't seems like a general fix, not CAVLC specific. It should be in a separate commit?

File: codec/decoder/plus/src/welsDecoderExt.cpp [hunk 3]
```diff
@@ -744,7 +744,7 @@ void CWelsDecoder::OutputStatisticsLog (SDecoderStatistics& sDecoderStatistics)
 
 DECODING_STATE CWelsDecoder::ReorderPicturesInDisplay (unsigned char** ppDst, SBufferInfo* pDstInfo) {
   if (pDstInfo->iBufferStatus == 1 && m_pDecContext->pSps->uiProfileIdc != 66
-      && m_pDecContext->pPps->bEntropyCodingModeFlag) {
+      /*&& m_pDecContext->pPps->bEntropyCodingModeFlag*/) {
```
Comment (thaytan) [inline codec/decoder/plus/src/welsDecoderExt.cpp]: I think it would be better to delete the bEntropyCodingModeFlag check entirely. It's just clutter now.

**Comment by sijchen** [comment]
close this for conflicts.
